src/main/java/com/hubspot/jinjava/util/ChunkResolver.java

```diff
@@ -105,7 +107,12 @@ public String resolveChunks() {
       .getThrowInterpreterErrors();
     try {
       interpreter.getContext().setThrowInterpreterErrors(true);
-      return String.join("", getChunk(null));
+      String expression = String.join("", getChunk(null));
+      if (JINJAVA_NULL.equals(expression)) {
+        // Resolved value of null as a string is ''.
+        return JINJAVA_EMPTY_STRING;
```

Comment:
This should be the only place where null needs to be swapped out for the empty string as this is the last step where the object must be resolved to a string. (That string could be presented in the final output of the render)

Reply:
See https://private.hubteam.com/opengrok/xref/all/github.com-HubSpot-jinjava%23master/src/main/java/com/hubspot/jinjava/lib/expression/DefaultExpressionStrategy.java#22 where null is replaced with empty when converting to a string.